kubectl-rook-ceph.sh

```diff
@@ -273,6 +276,97 @@ function run_purge_osd() {
     rook ceph osd remove --osd-ids=$1 --force-osd-removal=$force_removal"
 }
 
+####################################################################################################
+# 'kubectl rook-ceph debug commands
+####################################################################################################
+
+function update_deployment_spec_command(){
+  deployment_spec=$1
+  main_command='["sleep","infinity"]'
+  deployment_spec=$(jq .template.spec.containers[0].command="$main_command" <<< $deployment_spec)
+  deployment_spec=$(jq .template.spec.containers[0].args=[] <<< $deployment_spec)
+  echo "$deployment_spec" # for returning the updated deployment_spec
+}  
+
+function update_deployment_spec_image(){
+  deployment_spec=$1
+  alternalte_image=\"$2\"
+  deployment_spec=$(jq .template.spec.containers[0].image="$alternalte_image" <<< $deployment_spec)
+  echo "$deployment_spec" # for returning the updated deployment_spec
+}
+
+function verify_deployment_exist(){
+  if [[ $deployment_name != *"mon"* ]] && [[ $deployment_name != *"osd"* ]]; then
+    fail_error "only mon or osd deployment name can passed for starting debug mode"
+  fi
+  if _="$(KUBECTL_NS_CLUSTER get deployment "$deployment_name")"; then
+    echo "setting debug mode for \"$deployment_name\"" 
```

Comment:
This message is untrue when `verify_deployment_exist` is used from `run_stop_debug()`. I would remove this from the function and allow the caller to print a "good" message after this function returns successfully.

```sh
if ! KUBECTL_NS_CLUSTER get deployment "$deployment_name"; then
  fail_error "deployment $deployment_name doesn't exist"
fi
```

Reply:
I need to return some number value for it like 1-> there is no error, 2-> if there is error

The make a if else in main function as it is a shell function

I thought this will be a more complicated flow for reading, right?

Your thought?